wallet/wallet.go

```diff
@@ -1455,17 +1431,23 @@ func (w *Wallet) CalculateAccountBalance(account uint32, confirms int32,
 // CalculateAccountBalances calculates the values for the wtxmgr struct Balance,
 // which includes the total balance, the spendable balance, and the balance
 // which has yet to mature.
-func (w *Wallet) CalculateAccountBalances(account uint32, confirms int32) (wtxmgr.Balances, error) {
-	var balances wtxmgr.Balances
+func (w *Wallet) CalculateAccountBalances(confirms int32) (map[uint32]*wtxmgr.Balances, error) {
+	balances := make(map[uint32]*wtxmgr.Balances)
+
 	err := walletdb.View(w.db, func(tx walletdb.ReadTx) error {
 		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
 		txmgrNs := tx.ReadBucket(wtxmgrNamespaceKey)
-		var err error
-
-		balances, err = w.TxStore.AccountBalances(txmgrNs, addrmgrNs,
-			confirms, account)
-		return err
+		return w.Manager.ForEachAccount(addrmgrNs, func(acct uint32) error {
+			balance, err := w.TxStore.AccountBalance(txmgrNs, addrmgrNs,
```

Comment:
This really needs to be improved.  We should not be doing a "fullscan" balance on each account individually.  Instead we should do it once for all accounts.